lib/reply.js

```diff
@@ -159,7 +159,11 @@ Reply.prototype.header = function (key, value) {
 
   if (this[kReplyHeaders][_key] && _key === 'set-cookie') {
     // https://tools.ietf.org/html/rfc7230#section-3.2.2
-    this[kReplyHeaders][_key] = [this[kReplyHeaders][_key]].concat(value)
+    var _cookies = this[kReplyHeaders][_key]
+    if (typeof _cookies === 'string') {
+      _cookies = [_cookies]
+    }
+    this[kReplyHeaders][_key] = _cookies.concat(value)
```

Comment:
Agree

Reply:
Incoming value could be an array of cookies and we would need to merge them.

Reply:
I do not think an array is a good value for the “header()” method. We should probably throw in that case

Reply:
I agree with @mcollina. There is a `reply.headers()` method if one wants to pass in an array of values. The `reply.header()` method does not account for an array value in any case. Regardless, I'd still rather see the array updated instead of clobbered every time:

``js
// Update in-place if the value is an array
Array.prototype.push.apply(this[kReplyHeaders][_key], value)
```

Reply:
@mcollina I have seen some places where value could be an array for 'set-cookie' header. If we look at this code of 'fastify-reply-form', we are trying to copy the headers from a response to reply. 

https://github.com/fastify/fastify-reply-from/blob/36a3153d33a63d4031f5af41b3e0deeae3635291/lib/utils.js#L15-L27

At line 24, for 'set-cookie' header key, we would be getting an array as value which we are trying to set on reply. Let me know if I am missing something here.

@jsumners Thanks for sharing the code snippet. I will make the change to use push method.